File: 2-Rekursjon/hanoi.c
#include <stdio.h>

void hanoi(int n, char fra, char hjelp, char til) {
  if (n <= 1) {
    printf("%c->%c\n", fra, til);
  } 
  else {
    hanoi(n-1, fra, til, hjelp);
    printf("%c->%c\n", fra, til);
    hanoi(n-1, hjelp, fra, til);
  }
}

int main (int argc, char **argv) {
  int n = atoi(argv[1]);
  printf("Tårnet i Hanoi\n.");
  printf("Flytte ringene fra pinne A til pinne C, ved hjelp av pinne B:\n");
  hanoi(n, 'A', 'B', 'C');
}

File: 2-Rekursjon/Hanoi.java
public class Hanoi {

  //Antall ringer, pinner
  public static void hanoi(int n, char fra, char hjelper, char til) {
    if (n <= 1) System.out.println(fra + "->" + til);
    else {
      hanoi(n-1, fra, til, hjelper);
      System.out.println(fra + "->" + til);
      hanoi(n-1, hjelper, fra, til);
    }
 }

 public static void main(String[] args) {
    int n = Integer.parseInt(args[0]);
    System.out.println("Tårnet i Hanoi");
    System.out.println("Flytte "+ n + " ring(er) fra pinne A til C, ved hjelp av B");
    hanoi(n, 'A', 'B', 'C');
  }

}
